boto3/resources/factory.py

```diff
@@ -194,6 +194,17 @@ def _load_waiters(self, attrs, model):
         for waiter in model.waiters:
             attrs[waiter.name] = self._create_waiter(waiter)
 
+    def _create_identifier(factory_self, name):
+        """
+        Creates a read-only property for identifier attributes.
+        """
+        def identifier(self):
+            return getattr(self, '_' + name, None)
```

Comment:
Yeah I am fine with that. I do not think that it will happen unless there is a bug. Furthermore, since this is an identifier in general, its value cannot be ``None`` so it make sense not to default to ``None``.